### PR TITLE
admin activity log search: give a way to limit search period #4370

### LINKED ISSUE #4370: admin activity log search: give a way to limit search period
If the search period is not limited, it can go on searching which can drive up the bill.

### PR DESCRIPTION
**Pull request by thyageshm**
fixes #4370

### CODE REVIEW

**Comment by thyageshm** [comment]
@damithc I believe it is done. I checked it on the staging server. I am not sure who to assign this pr to.

**Comment by damithc** [comment]
@bamboo3250 ?

File: src/main/java/teammates/ui/controller/AdminActivityLogPageData.java [hunk 1]
```diff
@@ -276,15 +283,30 @@ private QueryParameters parseQuery(String query) throws Exception{
                     versions.add(values[j].replace(".", "-"));
                 }
                 
+            } else if (label.equals("from")) {
+                SimpleDateFormat sdf = new SimpleDateFormat("dd/MM/yy HH:mm");
+                Date d = sdf.parse(values[0] + " 00:00");                
+                Calendar cal = Calendar.getInstance();
+                cal.setTime(d);
+                cal = TimeHelper.convertToUserTimeZone(cal, - Const.SystemParams.ADMIN_TIMZE_ZONE_DOUBLE);
```
Comment (bamboo3250) [inline src/main/java/teammates/ui/controller/AdminActivityLogPageData.java]: Not from your PR but is this constant a typo?

File: src/main/java/teammates/ui/controller/AdminActivityLogPageData.java [hunk 2]
```diff
@@ -276,15 +283,30 @@ private QueryParameters parseQuery(String query) throws Exception{
...
+                fromDateValue = cal.getTime().getTime();
+                
+            } else if (label.equals("to")) {
+                SimpleDateFormat sdf = new SimpleDateFormat("dd/MM/yy HH:mm");
+                Date d = sdf.parse(values[0] + " 23:59");                
+                Calendar cal = Calendar.getInstance();
```
Comment (bamboo3250) [inline src/main/java/teammates/ui/controller/AdminActivityLogPageData.java]: Shouldn't you use `TimeHelper.now(Const.SystemParams.ADMIN_TIMZE_ZONE_DOUBLE)` here to make sure it get the correct time in Singapore? I think contributors in different time zone may get different value of the calendar.

File: src/main/java/teammates/ui/controller/AdminActivityLogPageData.java (hunk 1 above)
Comment (thyageshm) [inline src/main/java/teammates/ui/controller/AdminActivityLogPageData.java]: Yup! haha

File: src/main/java/teammates/ui/controller/AdminActivityLogPageData.java (hunk 2 above)
Comment (thyageshm) [inline src/main/java/teammates/ui/controller/AdminActivityLogPageData.java]: Now that I think about it, yes it makes sense to update this. I just copied this block of code from the prev implementation and moved it here haha

Comment (thyageshm) [inline src/main/java/teammates/ui/controller/AdminActivityLogPageData.java]: Hey actually, a few lines below the timezone is offset back to singapore time. I think the timezone is currently assumed to be in SGT. But anyway, I have updated the code

**Comment by thyageshm** [comment]
@bamboo3250 I think it should be fine now :-)

**Comment by bamboo3250** [comment]
@thyageshm ok, it is good to merge now

**Pull request merged**